File: 2025-12/src/day_10/boj_12779/Main.java
package day_10.boj_12779;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        long a = Long.parseLong(st.nextToken());
        long b = Long.parseLong(st.nextToken());

        long cnt = upperBound(b) - upperBound(a);

        if (cnt == 0) {
            System.out.println(0);
        } else {
            long gcd = gcd(cnt, b - a);
            System.out.printf("%d/%d", cnt / gcd, (b - a) / gcd);
        }
    }

    static long upperBound(long key) {
        long left = 0;
        long right = Integer.MAX_VALUE;

        while (left < right) {
            long mid = (left + right) / 2;

            if (mid * mid <= key) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return right;
    }

    static long gcd(long a, long b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
}

File: 2025-12/src/day_10/boj_12779/Main.cpp
#include <bits/stdc++.h>
using namespace std;

long long upper_bound_param(long long key) {
    long long left = 0;
    long long right = INT_MAX;

    while (left < right) {
        long long mid = (left + right) / 2;

        if (mid * mid <= key) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    return right;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    long long a, b;
    cin >> a >> b;

    long long cnt = upper_bound_param(b) - upper_bound_param(a);

    if (cnt == 0) {
        cout << 0;
    } else {
        long long g = gcd(cnt, b - a);
        cout << cnt / g << '/' << (b - a) / g;
    }
}
